src/Documents/Session/Operations/GetRevisionOperation.ts

```diff
@@ -0,0 +1,129 @@
+        if (!session) {
+            throwError("InvalidArgumentException", "Session cannot be null");
+        }
+
+        if (!changeVectorOrChangeVectorsOrId) {
+            throwError("InvalidArgumentException", "Id cannot be null");
+        }
+
+        this._session = session;
+        if (startOrDate instanceof Date) {
+            this._command = new GetRevisionsCommand(changeVectorOrChangeVectorsOrId as string, startOrDate);
+        } else if (TypeUtil.isArray(changeVectorOrChangeVectorsOrId)) {
+            this._command = new GetRevisionsCommand(changeVectorOrChangeVectorsOrId);
+        } else if (TypeUtil.isNumber(startOrDate)) {
+            this._command = new GetRevisionsCommand(changeVectorOrChangeVectorsOrId, startOrDate,
+                pageSize, metadataOnly);
+        } else {
+            this._command = new GetRevisionsCommand(changeVectorOrChangeVectorsOrId);
+        }
+    }
+
+    public createRequest() {
+        return this._command;
+    }
+
+    set result(result: IRavenArrayResult) {
+        this._result = result;
+    }
+
+    private _getRevision<TEntity extends object>(documentType: DocumentType<TEntity>, document: IRavenObject): TEntity {
+        if (!document) {
+            return null;
+        }
+
+        let id: string = null;
+
+        const metadata = document[CONSTANTS.Documents.Metadata.KEY];
+        if (metadata) {
+            id = metadata[CONSTANTS.Documents.Metadata.ID];
+        }
+
+        let changeVector = null as string;
+        if (metadata) {
+            changeVector = metadata[CONSTANTS.Documents.Metadata.CHANGE_VECTOR];
+        }
+
+        const entity = this._session.entityToJson.convertToEntity(documentType, id, document) as any as TEntity;
+        const documentInfo = new DocumentInfo();
+        documentInfo.id = id;
+        documentInfo.changeVector = changeVector;
+        documentInfo.document = document;
+        documentInfo.metadata = metadata;
+        documentInfo.entity = entity;
+        this._session.documentsByEntity.set(entity, documentInfo);
+        return entity;
+    }
+
+    public getRevisionsFor<TEntity extends object>(documentType: DocumentType<TEntity>): TEntity[] {
+        const resultsCount = this._result.results.length;
+        const results = [] as TEntity[];
+
+        for (const document of this._result.results) {
+            results.push(this._getRevision<TEntity>(documentType, document));
+        }
+        return results;
+    }
+
+    public getRevisionsMetadataFor(): MetadataAsDictionary[] {
+        const resultsCount = this._result.results.length;
+        const results = [] as MetadataAsDictionary[];
+
+        for (const document of this._result.results) {
+            const metadata = document[CONSTANTS.Documents.Metadata.KEY];
+            results.push(metadata);
+            //TODO: do we have to wrap this?   results.add(new MetadataAsDictionary(metadata)); (Greg)
```

Comment:
question is do we need to track changes on it and I don't think so, right?

Reply:
it returns revisions which are immutable, so it doesn't make sense to track changes.

Reply:
I've removed my comment then.